templates/Bulk_RNA_seq/Dockerfile

```diff
@@ -0,0 +1,33 @@
+# We will use Ubuntu for our image
+FROM ubuntu:latest
+
+ENV DEBIAN_FRONTEND "noninteractive"
+ENV TZ "America/New_York"
+
+RUN set -x \
+  && apt-get update \
+  && apt-get install -y\
+    tzdata \
+    python3 \
+    python3-pip \
+    git \
+    r-base \
+    make \
+    gcc
+
+WORKDIR /usr/src/app
+
+COPY setup.R ./
+RUN R -e "source('setup.R')"
+
+COPY requirements.txt ./
+RUN pip3 install --no-cache-dir -r requirements.txt
+
+COPY . .
+COPY ./biojupies_bulk2/ /usr/local/lib/python3.8/dist-packages/jupyter_template/profiles/biojupies_bulk/
+COPY run.sh ./
+RUN set +x run.sh
+
+EXPOSE 81
```

Comment:
You must use port 80 for it to integrate properly with the catalog

Reply:
furthermore, PREFIX environment must be modifyiable externally, using a script here makes injecting environment variables impossible. Please review this *automatically generated* Dockerfile which works with the example:

```
FROM ubuntu

RUN set -x \
    && echo "Preparing system..." \
    && apt-get -y update \
    && apt-get -y install git

ADD deps.txt /app/deps.txt
RUN set -x \
  && echo "Installing system dependencies from deps.txt..." \
  && apt-get -y update \
  && apt-get -y install $(grep -v '^#' /app/deps.txt) \
  && rm /app/deps.txt

ADD requirements.txt /app/requirements.txt
RUN set -x \
  && echo "Installing python..." \
  && apt-get install -y  python3-pip python3-dev \
  && pip3 install --upgrade pip \
  && echo "Installing python dependencies from requirements.txt..." \
  && pip3 install -Ivr /app/requirements.txt \
  && rm /app/requirements.txt

WORKDIR /app
EXPOSE 80

ENV PREFIX "/"
ENV HOST "0.0.0.0"
ENV PORT "80"

COPY . /app

CMD [ "jupyter-template", "--profile=biojupies", "example.ipynb" ]

```